docs/api/mcp/index.mdx

```diff
@@ -0,0 +1,273 @@
+# Sourcegraph MCP Server
+
+<p className="subtitle">Connect AI agents and applications to your Sourcegraph instance's code search and analysis capabilities.</p>
+
+The Sourcegraph Model Context Protocol (MCP) Server provides AI agents and applications with programmatic access to your Sourcegraph instance's code search, navigation, and analysis capabilities through a standardized interface.
+
+<Callout type="info">MCP Server requires Sourcegraph 6.8+ and is available on all deployment types.</Callout>
+
+## Server Endpoint
+
+The MCP server is available at:
+
+```
+https://your-sourcegraph-instance.com/.api/mcp/v1
+```
+
+## Authentication
+
+The Sourcegraph MCP server supports three authentication methods:
+
+### Access Token (Query Parameter)
```

Comment:
Should we recommend this? query parameters are generally considered not sensitive so those can even show up in Cloudflare logs and things 🙃 cc @evict if you have any thoughts on this.

Reply:
Let's not recommend that! It shows up in plain Cloudflare logs. We should probably look into removing this auth method entirely.

Reply:
Still not great to recommend this IMO, we should implement custom headers or OAuth in Amp then, there's currently nothing that treats the URLs as particularly sensitive I believe so easy to mishandle the URL containing the token. 

This feels like a useful investment generally to support other tools as well. :)

Reply:
Removed